src/device/mod.rs

```diff
@@ -233,11 +240,7 @@ pub trait Device {
     fn reset_state(&mut self);
 
     /// Submit a command buffer for execution
-    fn submit(&mut self, (
-        &Self::CommandBuffer,
-        &draw::DataBuffer,
-        &handle::Manager<Self::Resources>
-    ));
+    fn submit(&mut self, SubmitInfo<Self>);
```

Comment:
It would be inconvenient, since you wouldn't be able to do `device.submit(renderer.as_buffer())` any more.